race_test.go

```diff
@@ -0,0 +1,41 @@
+package smtpmock_test
+
+import (
+	"fmt"
+	"net/smtp"
+	"testing"
+	"time"
+
+	smtpmock "github.com/mocktools/go-smtp-mock/v2"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRace(t *testing.T) {
```

Comment:
Do we need an independent test file with different direct call for tracking race conditions? If not may better way to move it to `smtpmock_test.go`